feat: add an option to exit the dns-selector script

ssdrive-flaktra opened the pull request:
Resolves #1399

This adds an option to the `ujust dns-selector` script that just exits the script as requested by #1399.

Draft because I'm still testing it on my machine.

ssdrive-flaktra committed:
feat(dns-resolver): add an option to exit the script

ssdrive-flaktra commented:
I can confirm this works.

alexvojproc commented on files/system/usr/libexec/secureblue/dns_selector.py:
```diff
@@ -50,16 +50,16 @@
 
 def ask_option(options_count: int) -> int:
-    """Returns the user's chosen number between 1 and options_count."""
+    """Returns the user's chosen number between 0 and options_count."""
```
Why do you feel it's better to have a zero index across the whole script?

If you have a reason, remember you'll need to adjust the 7 other invocations of `ask_option()` and change the corresponding text.

alexvojproc replied:
e.g.
<img width="427" height="112" alt="image" src="https://github.com/user-attachments/assets/1bea4d7f-28f5-4146-96b9-fc89fc56103a" />

alexvojproc commented on files/system/usr/libexec/secureblue/dns_selector.py:
```diff
@@ -291,6 +293,10 @@ def run_interactive() -> int:
     match mode:
+        case 0:
+            # Exit
+            exit_code = 1
```
Why is being asked to exit an error condition?

alexvojproc replied:
Why do you set `exit_code = 1` again right after it's defined?

alexvojproc commented on files/system/usr/libexec/secureblue/dns_selector.py:
```diff
@@ -557,7 +563,8 @@ def main() -> int:
 
-    print_all_status()
+    if exit_code != 1:
+        print_all_status()
```
Remember the `exit_code`s come from `sandbox.run()`. If this returns non-zero, the transaction may not have been successful, leaving the system in an unknown state. Better to print status anyway.

alexvojproc reviewed (changes requested):
Fetching your branch (commit c62bc8e3725e41ed30acbbdd12a5c85460741e28), your code doesn't work for me.

<img width="504" height="500" alt="Please enter a number between 0 and 5." src="https://github.com/user-attachments/assets/2a51c6d5-a883-4b47-8932-7d5336f11352" />

alexvojproc commented:
It seems you might have used another GPG signing key that doesn't match your GitHub account, ideally just [switch it for this repo](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key) or add the email to your account. :) You're currently using 2FA8A8E4EAF8900A

ssdrive-flaktra committed:
fix(dns-selector): implement exiting in a better way

instead of setting the exit option to 0, we now just append it to the end of the list

ssdrive-flaktra committed:
lint(dns-selector): run ruff format and apply linter changes

ssdrive-flaktra committed:
fix(dns-selector): add a message to tell the user to hit ctrl+c to exit

seriously how did we not think of this?

ssdrive-flaktra committed:
lint: add newline
